packages/component-aws-s3/README.md

```diff
@@ -0,0 +1,227 @@
+# DBOS AWS Simple Storage Service (S3) Component
+
+This is a [DBOS](https://docs.dbos.dev/) library for working with [Amazon Web Services Simple Storage Service (S3)](https://aws.amazon.com/s3/).
+
+## Getting Started
+In order to store and retrieve objects in S3, it is necessary to:
+- Register with AWS, create an S3 bucket, and create access credentials. (See [Getting started with Amazon S3](https://aws.amazon.com/s3/getting-started/) in AWS documentation.)
+- If browser-based clients will read from S3 directly, set up S3 CORS.  (See [Using cross-origin resource sharing (CORS)](https://docs.aws.amazon.com/AmazonS3/latest/userguide/cors.html) in AWS documentation.)
+
+## Configuring a DBOS Application with AWS S3
+First, ensure that the DBOS S3 component is installed into the application:
+```
+npm install --save @dbos-inc/component-aws-s3
+```
+
+Second, ensure that the library class is imported and exported from an application entrypoint source file:
+```typescript
+import { S3Ops } from "@dbos-inc/component-aws-s3";
+export { S3Ops };
+```
+
+Third, place appropriate configuration into the [`dbos-config.yaml`](https://docs.dbos.dev/api-reference/configuration) file; the following example will pull the AWS information from the environment:
+```yaml
+application:
+  aws_s3_configuration: aws_config # Optional if the section is called `aws_config`
+  aws_config:
+    aws_region: ${AWS_REGION}
+    aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+    aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+```
+
+If a different configuration file section should be used for S3, the `aws_s3_configuration` can be changed to indicate a configuration section for use with S3.  If multiple configurations are to be used, the application code will have to name and configure them.
+
+The application will need at least one s3 bucket.  This can be placed in the `application` section of `dbos-config.yaml` also, but the naming key is to be established by the application.
+
+## Selecting A Configuration
+`S3Ops` is a configured class.  The AWS configuration (or config file key name) and bucket name must be provided when a class instance is created, for example:
+```typescript
+const myS3 = configureInstance(S3Ops, 'myS3Bucket', {awscfgname: 'aws_config', bucket: 'my-s3-bucket', ...});
+```
+
+## Simple Operation Wrappers
+The `S3Ops` class provides several [communicator](https://docs.dbos.dev/tutorials/communicator-tutorial) wrappers for S3 functions.
+
+### Reading and Writing S3 From DBOS Handlers and Workflows
+
+#### Writing S3 Objects
+A string can be written to an S3 key with the following:
+```typescript
+    const putres = await ctx.invoke(defaultS3).putS3Comm('/my/test/key', "Test string from DBOS");
```

Comment:
I think `S3put` (`S3get`...) would be cleaner names.

Reply:
What about leaving S3 off, it's implied from the class.

Reply:
That also works!